lua/xcodebuild/project/manager.lua

```diff
@@ -41,6 +41,76 @@ local function validate_xcodeproj_tool()
   return true
 end
 
+---Searches for a `.xcodeproj` file in the given directory.
+---
+---Uses the `find` command to search for the first `.xcodeproj` file
+---in the specified directory and returns its absolute path if found.
+---
+---@param dir string # The directory to search in.
+---@return string|nil # The absolute path to the `.xcodeproj` file, or nil if not found.
+local function findXcodeproj_file(dir)
+  local cmd = "find '" .. dir .. '\' -maxdepth 1 -name "*.xcodeproj" -print -quit 2> /dev/null'
+  local result = util.shell(cmd)
+  if result and #result > 0 and result[1] ~= "" then
+    return result[1]
+  else
+    return nil
+  end
+end
+
+---Traverses up the directory path to find a `.xcodeproj` file.
+---
+---Starting from the given path, this function moves up the directory
+---hierarchy searching for a `.xcodeproj` file in each directory up to cwd, at which point it stops.
+---Returns the absolute path to the `.xcodeproj` file if found.
+---
+---@param path string # The starting path to begin the search from.
+---@return string|nil # The absolute path to the `.xcodeproj` file, or nil if not found.
+local function findXcodeproj_path(path)
+  local dir = path
+  local cwd = vim.fn.getcwd()
+
+  while dir and dir ~= cwd and dir ~= "/" and dir ~= "" do
+    local xcodeproj = findXcodeproj_file(dir)
+    if xcodeproj then
+      if xcodeproj == projectConfig.settings.xcodeproj then
+        return nil
+      end
+
+      return xcodeproj
+    end
+
+    dir = dir:match("(.*/)[^/]+/?$")
+    if dir and dir:sub(-1) == "/" then
+      dir = dir:sub(1, -2)
+    end
+  end
+
+  return nil
+end
+
+--- Iterates over the provided parameters, checking if any parameter is a path.
+--- If a path is found, it searches for an `.xcodeproj` file starting from that parameter's path.
+--- Upon finding a `.xcodeproj` file, it inserts the path at the first position of the parameters
+--- list and returns `true`. If no `.xcodeproj` is found, returns `false`.
+---
+---@param table table # The table into which the `.xcodeproj` path is injected.
+---@param params string[] # The list of parameters to search through.
+---@return boolean # Returns true if an `.xcodeproj` path is found and injected; otherwise, false.
+local function inject_relative_xcodeproj(table, params)
+  for _, param in ipairs(params) do
+    if string.find(param, "/") then
```

Comment:
I think it would be best to check here `vim.startswith(param, "/")` instead to make sure that it's actually path